src/coprocessor/codec/convert.rs

```diff
@@ -1,14 +1,55 @@
 // Copyright 2016 TiKV Project Authors. Licensed under Apache-2.0.
 
 use std::borrow::Cow;
-use std::{self, i64, str, u64};
+use std::{self, char, i16, i32, i64, i8, str, u16, u32, u64, u8};
 
 use cop_datatype::{self, FieldTypeTp};
 
-use super::mysql::Res;
+use super::mysql::{Res, RoundMode, DEFAULT_FSP};
 use super::{Error, Result};
+use crate::coprocessor::codec::data_type::{DateTime, Decimal, Duration, Json};
+use crate::coprocessor::codec::error::ERR_DATA_OUT_OF_RANGE;
 use crate::coprocessor::dag::expr::EvalContext;
 
+/// `integer_unsigned_upper_bound` returns the max u64 values of different mysql types
+#[inline]
+pub fn integer_unsigned_upper_bound(tp: FieldTypeTp) -> u64 {
+    match tp {
+        FieldTypeTp::Tiny => u64::from(u8::MAX),
+        FieldTypeTp::Short => u64::from(u16::MAX),
+        FieldTypeTp::Int24 => (1u64 << 24) - 1u64,
```

Comment:
what about simply 1 << 24 - 1

Reply:
Do we have any preference for similar writing in the TiKV?

Reply:
nop, but 1 << 24 - 1 looks very clear and simple

Reply:
OK, got it.